src/cli/self_update/windows.rs

```diff
@@ -194,9 +194,9 @@ fn _apply_new_path(new_path: Option<String>) -> Result<()> {
 }
 
 // Get the windows PATH variable out of the registry as a String. If
-// this returns None then the PATH variable is not unicode and we
+// this returns None then the PATH variable is not a string and we
 // should not mess with it.
-fn get_windows_path_var() -> Result<Option<String>> {
+fn get_windows_path_var() -> Result<Option<Vec<u16>>> {
```

Comment:
OsString again: OsString on windows can represent any u16 string I believe, though if I'm wrong, I'd like to know. See this trait to help with this particular code path: https://doc.rust-lang.org/stable/std/os/windows/ffi/trait.OsStringExt.html